modules/ROOT/pages/queries/expressions.adoc

```diff
@@ -0,0 +1,79 @@
+= Cypher expressions
+:description: This page explains which expressions are allowed in Cypher.
+
+This page contains examples of allowed expressions in Cypher.
+
+[NOTE]
+====
+Most expressions in Cypher evaluate to `null` if any of their inner expressions are `null`.
+Notable exceptions are the operators `IS NULL`, `IS NOT NULL`, and the xref:values-and-types/type-predicate.adoc[type predicate expressions].
+====
+
+[[general]]
+== General
+
+* A variable: `n`, `x`, `rel`, `myFancyVariable`, `++`A name with special characters in it[]!`++`.
+* A property: `n.prop`, `x.prop`, `rel.thisProperty`, `++myFancyVariable.`(special property name)`++`.
+* A dynamic property: `n["prop"]`, `rel[n.city + n.zip]`, `map[coll[0]]`.
+* A parameter: `$param`, `$0`.
+* A list of expressions: `['a', 'b']`, `[1, 2, 3]`, `['a', 2, n.property, $param]`, `[]`.
+* A function call: `length(p)`, `nodes(p)`.
+* An aggregate function call: `avg(x.prop)`, `+count(*)+`.
+* A path-pattern: `+(a)-[r]->(b)+`, `+(a)-[r]-(b)+`, `+(a)--(b)+`, `+(a)-->()<--(b)+`.
+* An operator application: `1 + 2`, `3 < 4`.
+* A subquery expression: `COUNT {}`, `COLLECT {}`, `EXISTS {}`, `CALL {}`.
+* A regular expression: `a.name =~ 'Tim.*'`.
+* A `CASE` expression.
+
+[NOTE]
+====
+Using regular expressions with unsanitized user input may make your application vulnerable to Cypher injection.
+Consider using xref:syntax/parameters.adoc[parameters] instead.
+====
```

Comment:
Why do we have this note on regexes? Unsanitized user input is bad in all expressions, not just regexes. I don't understand what information we're trying to convey here.

Reply:
@gem-neo4j any reason why it should stay?

Reply:
This came from this card: https://trello.com/c/u2A4QRdw/468-regex-escape-clarification to do with security issues, so, yes unsanitized user input is bad in all expressions, so perhaps this could be broadened instead of just removed?

Reply:
Yeah, and also link to the KB article you co-authored!